pkg/certificate/manager.go

```diff
@@ -16,30 +17,79 @@ var (
 	errCertNotFound = errors.New("certificate not found")
 )
 
+const checkCertificateExpirationInterval = 5 * time.Second
+
 // NewManager creates a new CertManager with the passed CA and CA Private Key
 func NewManager(
 	ca *Certificate,
 	client client,
 	serviceCertValidityDuration time.Duration,
-	msgBroker *messaging.Broker) (*manager, error) { //nolint:revive // unexported-return
+	msgBroker *messaging.Broker,
+	certRotation <-chan struct{}) (*Manager, error) {
 	if ca == nil {
 		return nil, errNoIssuingCA
 	}
 
-	m := &manager{
+	m := &Manager{
 		// The root certificate signing all newly issued certificates
 		ca:                          ca,
 		client:                      client,
 		serviceCertValidityDuration: serviceCertValidityDuration,
 		msgBroker:                   msgBroker,
+		certRotation:                certRotation,
 	}
+	m.start(checkCertificateExpirationInterval)
+	return m, nil
+}
 
-	// TODO(#4533) start the cert rotation here.
+func (m *Manager) start(checkInterval time.Duration) {
+	// iterate over the list of certificates
+	// when a cert needs to be rotated - call RotateCertificate()
+	ticker := time.NewTicker(checkInterval)
```

Comment:
piggybacking jon's suggestion let's have a call to m.checkAndRotate to start this off

Reply:
you can put it in the goroutine, prior to the for loop